code/game/objects/structures/flora.dm

```diff
@@ -73,14 +73,106 @@
 	..()
 	icon_state = "snowbush[rand(1, 6)]"
 
+
+
+
+//Potted Plants
 /obj/structure/flora/pottedplant
 	name = "potted plant"
 	icon = 'icons/obj/plants.dmi'
 	icon_state = "plant-26"
+	var/dead = 0
+	var/obj/item/stored_item
 	layer = ABOVE_MOB_LAYER
 
-//newbushes
 
+/obj/structure/flora/pottedplant/Destroy()
+	QDEL_NULL(stored_item)
+	return ..()
+
+/obj/structure/flora/pottedplant/proc/death()
+	if (!dead)
+		icon_state = "plant-dead"
+		name = "dead [name]"
+		desc = "It looks dead."
+		dead = 1
+//No complex interactions, just make them fragile
+/obj/structure/flora/pottedplant/ex_act(var/severity = 2.0)
+	death()
+	return ..()
+
+/obj/structure/flora/pottedplant/fire_act()
+	death()
+	return ..()
+
+/obj/structure/flora/pottedplant/attackby(obj/item/W, mob/user)
+	user.visible_message("[user] begins digging around inside of \the [src].", "You begin digging around in \the [src], trying to hide \the [W].")
+	if(do_after(user, 20, target = src))
+		if(!stored_item)
+			if(W.w_class <= 3)
+				user.drop_from_inventory(W,src)
```

Comment:
`drop_from_inventory` is unconditional. So robots can drop modules and humans can drop undropable stuff (like tools from prosthesis)

Reply:
I've added in a check for if mob.can_unequip which should be sufficient to prevent any issues

Reply:
Where exactly? I can't see it.